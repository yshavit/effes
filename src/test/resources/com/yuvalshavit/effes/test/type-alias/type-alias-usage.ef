type Boolean = True | False

def invert (b: Boolean) -> Boolean:
  return case b of
    True: False
    False: True

def main:
  print (invert True)
  print (invert False)
