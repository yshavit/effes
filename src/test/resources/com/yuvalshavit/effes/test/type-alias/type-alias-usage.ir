def invert (b: False | True) -> False | True:
  ReturnStatement
    CaseExpression
      case:
        VarExpression
          arg: b
          pos: 0
      of True:
        CtorInvoke
          type: False
      of False:
        CtorInvoke
          type: True
