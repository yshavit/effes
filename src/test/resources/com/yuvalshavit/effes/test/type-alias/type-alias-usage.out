Your argument was: False
Your argument was: True
>> null
